testplan/testing/multitest/driver/connection/base.py

```diff
@@ -0,0 +1,69 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Union
+from collections import defaultdict
+
+from testplan.common.utils.context import ContextValue
+
+
+class Direction(Enum):
+    CONNECTING = "connecting"
+    LISTENING = "listening"
+
+
+@dataclass
+class BaseConnectionInfo:
```

Comment:
I suggest that we rename this series of class to ConnectionExtractor - BaseConnectionExtractor, PortConnectionExtractor, FileConnectionExtractor etc.

Add a new ConfigOption connection_extractors to driver class, which takes a list of extractor objects. And a reasonable default for stock drivers.

The BaseConnectionExtractor is just going to read host/port information from driver's properties.
PortConnectionExtractor will check tcp/udp port etc.
And FileConnectionExtractor will check open fd.
(I think you are already doing above).

Let's decommission all driver metadata related feature from our code.

Create DriverConnections class to aggregate all connections of drivers in this environment.
(I see currently we are only using a list for that, I would prefer to have a proper class with necessary methods)